File: Leetcode/Trees/1302. Deepest Leaves Sum.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int deepestLeavesSum(TreeNode root) {
        int depth = Depth(root), currentHeight = 1;
        return CalculateSum(root, depth, currentHeight);
    }
    private int Depth(TreeNode Root) {
        if (Root == null) return 0;
        else return 1 + Math.max(Depth(Root.left), Depth(Root.right));
    }
    private int CalculateSum(TreeNode root, int depth, int currentHeight) {
        if (root == null) return 0;
        else if (currentHeight == depth) return root.val;
        else return CalculateSum(root.left, depth, currentHeight + 1) + CalculateSum(root.right, depth, currentHeight + 1);
    }
}

File: Leetcode/Trees/1302. Deepest Leaves Sum.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def deepestLeavesSum(self, root: Optional[TreeNode]) -> int:

        def Height(root):
            if root is None: 
                return 0
            else: 
                return 1 + max(Height(root.left), Height(root.right))
                
        def CalculateSum(root, depth, currentHeight):
            if root is None:
                return 0
            elif currentHeight == depth:
                return root.val
            else:
                return CalculateSum(root.left, depth, currentHeight + 1) + CalculateSum(root.right, depth, currentHeight + 1)
        
        return CalculateSum(root, Height(root), 1)